consensus/src/threshold_simplex/engine.rs

```diff
@@ -18,7 +18,8 @@ use tracing::debug;
 
 /// Instance of `threshold-simplex` consensus engine.
 pub struct Engine<
-    E: Clock + GClock + Rng + CryptoRng + Spawner + Storage + Metrics,
+    St: Storage,
+    E: Clock + GClock + Rng + CryptoRng + Spawner + Metrics,
```

Comment:
I prefer to keep all `E` references (i.e. "Context") as the first trait listed here.

Reply:
done